### PR TITLE
Handle the special case for “new visitors” and “returning visitors” to retrieve their metrics for the Selection Panel

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #8160 

## Relevant technical choices

I made a couple of small out-of-scope fixes, and also added JS coverage rather than an additional story. Please see the comments below for more detail on these.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js [hunk 1]
```diff
@@ -207,6 +207,7 @@ function AudienceSegmentationSetupCTAWidget( { Widget, WidgetNull } ) {
 	}
 
 	if (
+		configuredAudiences === undefined ||
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:210]: I noticed the Setup CTA Banner can flicker while this variable is being resolved. Checking for `undefined` here fixes it.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js [hunk 2]
```diff
@@ -230,6 +238,141 @@ describe( 'AudienceSelectionPanel', () => {
 				} );
 		} );
 
+		it.each( [
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js:241]: While working on the issue, I realised that providing an additional story as per the IB wouldn't really add much value, as there would be no visual difference to the story.

I therefore took the approach of adding test coverage for the changes. This test is based on the one above it.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceItems.js [hunk 3]
```diff
@@ -55,35 +55,103 @@ export default function AudienceItems( { savedItemSlugs = [] } ) {
 			return [];
 		}
 
+		// eslint-disable-next-line @wordpress/no-unused-vars-before-return -- We might return before `otherAudiences` is used.
+		const [ siteKitAudiences, otherAudiences ] = audiences.reduce(
+			( [ siteKit, other ], audience ) => {
+				if ( audience.audienceType === 'SITE_KIT_AUDIENCE' ) {
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceItems.js:61]: There's no need to use the `isSiteKitAudience()` selector here, seeing as we have access to `audienceType`. We do compare it directly elsewhere in the codebase, so this is a consistent approach.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:210]: Excellent, thanks for fixing it, @techanvil 👍

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceItems.js (hunk 3 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/AudienceItems.js:61]: Nice one 👍

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js (hunk 2 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSelectionPanel/index.test.js:241]: Makes sense 👍

**Review by hussain-t — APPROVED** [review]
Excellent work, @techanvil, thanks! LGTM 🎖️

**Review by nfmohit — APPROVED** [review]
LGTM ✅

**Pull request merged**